lib/puppet.rb

```diff
@@ -130,8 +134,14 @@ def self.run_mode
 
   # Now that settings are loaded we have the code loaded to be able to issue
   # deprecation warnings. Warn if we're on a deprecated ruby version.
-  if Gem::Version.new(RUBY_VERSION.dup) < Gem::Version.new(Puppet::OLDEST_RECOMMENDED_RUBY_VERSION)
-    Puppet.deprecation_warning(_("Support for ruby version %{version} is deprecated and will be removed in a future release. See https://puppet.com/docs/puppet/latest/system_requirements.html for a list of supported ruby versions.") % { version: RUBY_VERSION })
+  if defined?(JRUBY_VERSION)
+    if Gem::Version.new(JRUBY_VERSION) < Gem::Version.new(Puppet::OLDEST_RECOMMENDED_JRUBY_VERSION)
+      Puppet.deprecation_warning(_("Support for JRuby version %{version} is deprecated and will be removed in a future release.") % { version: JRUBY_VERSION })
+    end
```

Comment:
I'm not sure if we want to do this right now or wait until we actually have puppetserver builds with JRuby 9.4? Thoughts @justinstoller @mhashizume ?

Reply:
I dropped this and filed https://tickets.puppetlabs.com/browse/PUP-11716